model/host/host.go

```diff
@@ -1413,10 +1415,33 @@ func FindHostsToTerminate() ([]Host, error) {
 			{ // hosts that failed to provision
 				StatusKey: evergreen.HostProvisionFailed,
 			},
-			{ // hosts that are taking too long to provision
-				"$or": []bson.M{
-					bson.M{ProvisionedKey: false},
-					bson.M{StatusKey: evergreen.HostProvisioning},
+			{
```

Comment:
I'm a little concerned that this isn't covered by an index. Could you spot check this?

Reply:
The check for provisioned and status is "provisioning" (L1424-1425) are in production and haven't changed as far as I'm aware, so if it's not covered by an index at this point, then adding one won't change anything. The only thing I added is an additional check on user data hosts that are not provisioned.

There's a check on L1447 for not terminated status.

Reply:
SGTM